org.eclipse.jdt.core.manipulation/core extension/org/eclipse/jdt/internal/corext/refactoring/code/makestatic/InstanceUsageRewriter.java

```diff
@@ -236,15 +230,22 @@ private void modifyInstanceMethodUsage(SimpleName node, IBinding binding) {
 		}
 	}
 
-	private boolean isInsideAnonymousClass(ITypeBinding declaringClass) {
-		if (declaringClass != null) {
-			boolean isLocal= declaringClass.isLocal();
-			boolean isAnonymous= declaringClass.isAnonymous();
-			boolean isMember= declaringClass.isMember();
-			boolean isNested= declaringClass.isNested();
-			boolean isTopLevel= declaringClass.isTopLevel();
-			if (!isTopLevel || isNested || isAnonymous || isLocal || isMember) {
-				return true;
+	private boolean isAccessToAnonymousClass(ASTNode node, ITypeBinding declaringClass) {
+		ITypeBinding typeOfRefactoredMethod = fTargetMethodDeclaration.resolveBinding().getDeclaringClass();
```

Comment:
You have to verify that resolveBinding() does not return null before calling getDeclaringClass()

Reply:
Fixed with a690054.

Reply:
Thanks.   Marking this as resolved.